crates/isograph_compiler/src/isograph_literals.rs

```diff
@@ -110,45 +102,51 @@ fn visit_dirs_skipping_isograph(dir: &Path, cb: &mut dyn FnMut(&DirEntry)) -> io
 // both valid and invalid iso literals.
 #[allow(clippy::type_complexity)]
 pub fn parse_iso_literals_in_file_content(
-    file_path: PathBuf,
+    relative_path_to_source_file: RelativePathToSourceFile,
     file_content: String,
-    canonicalized_root_path: &Path,
-    config: &CompilerConfig,
+    current_working_directory: CurrentWorkingDirectory,
 ) -> Result<
-    (
-        RelativePathToSourceFile,
-        Vec<(IsoLiteralExtractionResult, TextSource)>,
-    ),
+    Vec<(IsoLiteralExtractionResult, TextSource)>,
     Vec<WithLocation<IsographLiteralParseError>>,
 > {
-    let absolute_path = canonicalized_root_path.join(&file_path);
-    let relative_path_to_source_file = relative_path_from_absolute_and_working_directory(
-        config.current_working_directory,
-        &absolute_path,
-    );
-
     let mut extraction_results = vec![];
     let mut isograph_literal_parse_errors = vec![];
 
     for iso_literal_extraction in extract_iso_literals_from_file_content(&file_content) {
         match process_iso_literal_extraction(
             iso_literal_extraction,
             relative_path_to_source_file,
-            relative_path_to_source_file,
-            config,
+            current_working_directory,
         ) {
             Ok(result) => extraction_results.push(result),
             Err(e) => isograph_literal_parse_errors.push(e),
         }
     }
 
     if isograph_literal_parse_errors.is_empty() {
-        Ok((relative_path_to_source_file, extraction_results))
+        Ok(extraction_results)
     } else {
         Err(isograph_literal_parse_errors)
     }
 }
 
+#[allow(clippy::type_complexity)]
+#[memo]
+pub fn parse_iso_literal_in_source(
+    db: &Database,
+    iso_literals_source_id: SourceId<IsoLiteralsSource>,
+    current_working_directory: CurrentWorkingDirectory,
+) -> Result<
+    Vec<(IsoLiteralExtractionResult, TextSource)>,
+    Vec<WithLocation<IsographLiteralParseError>>,
+> {
+    let IsoLiteralsSource {
+        relative_path,
+        content,
+    } = db.get(iso_literals_source_id).clone();
```

Comment:
nit: no need to clone here, we can just pass content as an &str, and relative impls copy